core/core_test.go

```diff
@@ -109,11 +110,27 @@ func TestDefaultAPI_CurrentUser(t *testing.T) {
 func TestDefaultAPI_TokenRenew(t *testing.T) {
 	category.Set(t, category.Integration)
 
+	idempotencyKey := uuid.New()
 	tests := []testCase{
 		testNewCase(t, http.StatusCreated, TokenRenewURL, "core_token_renew", nil),
 		testNewCase(t, http.StatusBadRequest, TokenRenewURL, "core_token_renew", ErrBadRequest),
 		testNewCase(t, http.StatusNotFound, TokenRenewURL, "core_token_renew", ErrNotFound),
 		testNewCase(t, http.StatusInternalServerError, TokenRenewURL, "", ErrServerInternal),
+
+		{
+			name: "Idempotent request",
+			handler: func(rw http.ResponseWriter, r *http.Request) {
+				assert.Equal(t, idempotencyKey.String(), r.Header.Get("Idempotency-Key"))
```

Comment:
Couldn't this go into the `testNewCase` and then it is checked for all the requests?

Reply:
@mariusSincovici 

It looks like `testNewCase()` is being used for other requests as well, which are not necessarily idempotent, thus testing for the existence of idempotency-key won't work.

It'd need a modification in `testNewCase()`, which I think doesn't make sense, and would result in even more code and complexity, than this simple `testCase{}` that is manually created.